torch/utils/data/datapipe.py

```diff
@@ -0,0 +1,199 @@
+import functools
+from typing import Dict, Callable, Optional
+
+from torch.utils.data._typing import _DataPipeMeta
+from torch.utils.data._utils.serialization import serialize_fn, SerializationType, deserialize_fn
+from torch.utils.data.dataset import IterableDataset, T_co, UNTRACABLE_DATAFRAME_PIPES, Dataset
```

Comment:
`T_co` and `UNTRACABLE_DATAFRAME_PIPES` should be created from this file rather than importing them.

And, if we do `import torch.utils.data.dataset.IterableDataset` and `import torch.utils.data.dataset.Dataset`, would circular import still happen?

Reply:
I just refactored it, though most of the import statements for DataPipes have to be changed.

I am explicitly leaving `graph.py` and `graph_settings.py` in `torch/utils/data` (instead of moving them into the `datapipes` directory) because they are needed by `DataLoader`. Let me know if you think those two files should be moved as well.